pkg/controller/namespaceconfig/namespaceconfig_controller.go

```diff
@@ -282,28 +282,13 @@ func (r *ReconcileNamespaceConfig) applyConfigToNamespace(objs []unstructured.Un
 }
 
 func createOrUpdate(client *dynamic.ResourceInterface, obj *unstructured.Unstructured) error {
-
-	obj2, err := (*client).Get(obj.GetName(), metav1.GetOptions{})
-
-	if apierrors.IsNotFound(err) {
-		_, err = (*client).Create(obj, metav1.CreateOptions{})
-		if err != nil {
-			log.Error(err, "unable to create object", "object", obj)
-		}
-		return err
-	}
-	if err == nil {
-		obj.SetResourceVersion(obj2.GetResourceVersion())
-		_, err = (*client).Update(obj, metav1.UpdateOptions{})
-		if err != nil {
-			log.Error(err, "unable to update object", "object", obj)
+	_, err := (*client).Create(obj, metav1.CreateOptions{})
+	if err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			_, err = (*client).Update(obj, metav1.UpdateOptions{})
```

Comment:
@etsauer For any update, you need to ensure that it has the proper `resourceVersion`. To ensure you have the latest version, you will need to query the most update to date object using the steps previously implemented:

```
obj2, err := (*client).Get(obj.GetName(), metav1.GetOptions{})
```

Then just prior to the update

```
obj.SetResourceVersion(obj2.GetResourceVersion())
```